boto/endpoints.json

```diff
@@ -451,9 +495,10 @@
     "s3": {
         "ap-northeast-1": "s3-ap-northeast-1.amazonaws.com",
         "ap-northeast-2": "s3.ap-northeast-2.amazonaws.com",
-        "ap-south-1": "s3-ap-south-1.amazonaws.com",
```

Comment:
Looks like the endpoint using the dot instead of the dash is valid, so should be fine that this changes.